manager/controlapi/cluster.go

```diff
@@ -58,6 +58,10 @@ func validateClusterSpec(spec *api.ClusterSpec) error {
 		}
 	}
 
+	if spec.Annotations.Name != store.DefaultClusterName {
+		return grpc.Errorf(codes.PermissionDenied, "modification of cluster name is not allowed")
```

Comment:
codes.InvalidArgument perhaps?

Reply:
I considered that, but I think the definition of PermissionDenied  is "PermissionDenied indicates the caller does not have permission to execute the specified operation.", which fits IMO.

Reply:
hm, feels a bit odd, because caller has access to this endpoint, but passed an invalid argument (The value was not accepted, because the only valid value here is "default"). I don't think it's any different than the other validations performed in this function (e.g. you cannot set any value other than "bcrypt" here; https://github.com/anshulpundir/swarmkit/blob/21e99c146ef0fce96ee89befb950bc75f3f73e68/manager/controlapi/cluster.go#L44

Reply:
Not sure I follow. Why do you think having access to an endpoint but passing invalid arg is odd ? @thaJeztah

Reply:
That's not odd, but returning a permission denied instead of invalid argument (like for all the other validation check in this function) feels odd.

Reply:
agreed. I already changed it to invalid argument. LMK if this looks good @thaJeztah @nishanttotla

Reply:
I agree with using `codes.InvalidArgument`.